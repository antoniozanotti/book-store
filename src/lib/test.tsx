import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
