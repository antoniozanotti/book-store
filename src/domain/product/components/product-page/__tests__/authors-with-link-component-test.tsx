import { render, screen } from "@/lib/test";
import { AuthorsWithLinkComponent } from "../authors-with-link-component";
import { authorsWithPageMock } from "@/domain/author/__tests__/authors-with-page-mock";

describe("AuthorsWithLinkComponent", () => {
  it("should render", async () => {
    const { container } = render(<AuthorsWithLinkComponent authors={authorsWithPageMock} />);
    expect(container).toMatchSnapshot();
  });
});