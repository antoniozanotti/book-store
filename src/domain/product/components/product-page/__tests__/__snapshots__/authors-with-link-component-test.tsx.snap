// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorsWithLinkComponent should render 1`] = `
<div>
  <div
    data-testid="container"
  >
    <address>
      by 
      <a
        class="underline"
        href="/author/kevin-kwan"
        rel="author"
        title="Kevin Kwan"
      >
        Kevin Kwan
      </a>
       and 
      <a
        class="underline"
        href="/author/julia-cameron"
        rel="author"
        title="Julia Cameron"
      >
        Julia Cameron
      </a>
       and 
      <a
        class="underline"
        href="/author/david-byrne"
        rel="author"
        title="David Byrne"
      >
        David Byrne
      </a>
       and 
      <a
        class="underline"
        href="/author/donna-tartt"
        rel="author"
        title="Donna Tartt"
      >
        Donna Tartt
      </a>
       and 
      <a
        class="underline"
        href="/author/jeremy-mccarter"
        rel="author"
        title="Jeremy McCarter"
      >
        Jeremy McCarter
      </a>
       and 
      <a
        class="underline"
        href="/author/linmanuel-miranda"
        rel="author"
        title="Lin-Manuel Miranda"
      >
        Lin-Manuel Miranda
      </a>
    </address>
  </div>
</div>
`;
